app/models/reference_repository.rb

```diff
@@ -9,6 +9,15 @@ class ReferenceRepository < ApplicationRecord
   before_save :force_index
 
   validates_uniqueness_of :re3doi, allow_nil: true
+  #
```

Comment:
I'm not sure whether this is part of this API but, how is one ReferenceRepository instace created when someone create a Client/Repo in Fabrica/REST-API?   

I see ReferenceRepository.find_or_create_by is triggered in rake tasks and in rspec but when running in production how are the new repositories added to the ReferenceRepository index/db? 

Sorry If I'm missing something.

Reply:
Fair point.  You are not missing anything.  It does not currently auto-update or auto-add on save.

Even though this PR adds a lot, I was planning on making sure the bones were in place before refining the creation/update steps an a future PR.  If that sounds OK, I can add a GH issue to make sure that intention is recorded.

Reply:
well I guess it's just a callback on create like in:

https://github.com/datacite/lupo/blob/449db11c9eec14573b80a6fd7918c74dee02af6e/app/models/doi.rb#L129

Reply:
In our zoom you showed that it wouldn't be a large lift to add this in.  Will do and update this PR.